Source/CombatExtended/Harmony/Harmony_FloatMenuMakerMap.cs

```diff
@@ -218,6 +218,28 @@ static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> inst
             string targetString = "CannotPickUp";
             int stringIndex = -1;
             List<CodeInstruction> codes = Modify_ForceWear(instructions).ToList();
+            var formingCaravan = AccessTools.Method(typeof(RimWorld.Planet.CaravanFormingUtility), "IsFormingCaravan");
+            bool foundCaravanInjection = false;
+            // Sets IsFormingCaravan to true for if(!pawn.IsFormingCaravan)
+            // Fixes duplicate float menu while not at home
+            for (int i = 0; i < codes.Count - 5; i++)
+            {
+                if (
+                    codes[i].opcode == OpCodes.Endfinally &&
+                    codes[i + 1].opcode == OpCodes.Ldloc_0 &&
+                    codes[i + 3].Calls(formingCaravan) &&
+                    codes[i + 4].opcode == OpCodes.Brtrue &&
+                    codes[i + 5].opcode == OpCodes.Ldloc_0)
+                {
+                    codes[i + 3].opcode = OpCodes.Ldc_I4_1;
+                    codes[i + 3].operand = null;
+                    foundCaravanInjection = true;
```

Comment:
Would be better to `break;` here after finding the correct target, to avoid continuing the search (and potentially finding an invalid second target when RW upstream updates). Otherwise, it's probably worth logging the number found, so if it changes we know something is up.